install.sh

```diff
@@ -329,6 +307,10 @@ do_install() {
 			if [ -z "$dist_version" ] && [ -r /etc/lsb-release ]; then
 				dist_version="$(. /etc/lsb-release && echo "$DISTRIB_CODENAME")"
 			fi
+			# Install curl if it's not already installed since it's important for downloading
+			if ! command_exists curl; then
+				(set -ex; $sh_c "apt-get update && apt-get install -y curl")
```

Comment:
fedora? centos?
`which curl || ( apt-get update && apt-get install -y curl) || yum install -y curl || zypper install -y curl || urpmi --auto curl || pacman --noconfirm -Sy curl || upgradepkg --install-new || apk --no-cache add curl`

Reply:
Good catch, but wow that solution feels messy. It looks like it would work, and as much as I love one liners when working in CLI, a script has a little more room to let code be code. Another thought is you've included things like pacman but this script doesn't support Arch for the moment. Maybe a more comprehensive PR to add full support for Arch would make more sense.

Reply:
Maybe something like this
```bash
if command_exists apt-get; then
    (set -ex; $sh_c "apt-get install -y curl")
elif command_exists dnf; then
    (set -ex; $sh_c "dnf install -y curl")
elif command_exists yum; then
    (set -ex; $sh_c "yum install -y curl")
fi
```

Reply:
You'll only reach this if you're running an old version of debian so I think the usage of `apt-get` is appropriate here

Reply:
So aside from an old version of debian all other supported distros already include curl?

Reply:
Oh wait nvm, I was confused this is only installation instructions for `ubuntu`,

<img width="838" alt="screen shot 2019-01-23 at 9 33 18 am" src="https://user-images.githubusercontent.com/1700823/51625433-ff425680-1ef1-11e9-8e7e-c5c84f16d1a2.png">

Reply:
Wow... That's what happens when we don't scroll up...

Anyway would it be appropriate for there to be similar lines in the other distros? Or do they all include curl?